src/scala/microsoft-spark-2-3/src/main/scala/org/apache/spark/deploy/dotnet/DotnetRunner.scala

```diff
@@ -143,6 +143,12 @@ object DotnetRunner extends Logging {
         } else {
           logInfo(s".NET application exited successfully")
         }
+
+        // Shutdown the thread pool whose executors could still be running,
```

Comment:
should this be moved in DotnetBackend.close()` ?

Reply:
That works with the current code, but the level is a bit different. If we allow multiple `DotnetBackend` objects, we shouldn't shutdown the ThreadPool (singleton), and that's why I put it outside.

But, I think it's already broken with `DotnetBackend.shutdownCallbackClient()`, so I will move it so that we can fix it at the same time if needed.